static/js/services/base64-file-reader.js

```diff
@@ -0,0 +1,22 @@
+angular.module('inboxServices').service('Base64FileReader', [
+  '$q',
+  function($q) {
+    'use strict';
+
+    return function(blob) {
+      var deferred = $q.defer();
+      var reader = new FileReader();
+      reader.addEventListener('loadend', function() {
+        deferred.resolve(reader.result);
+      });
+      reader.addEventListener('error', function() {
+        deferred.reject(reader.error);
+      });
+      reader.addEventListener('abort', function() {
+        deferred.reject(new Error('FileReader aborted.'));
+      });
+      reader.readAsDataURL(blob);
+      return deferred.promise;
+    };
+  }
+]);
```

Comment:
This is almost identical to the FileReader service. Find a way to share the code. I think my preferred solution would be to have FileReader take an `options` parameter with `base64:true` as an option. Then it's just adding a switch to the existing service rather than a whole new service.

Reply:
I've been pondering how to do this, as general usage is:
```
  .then(FileReader)
```

One idea would be for the service to return e.g. `{ base64:..., text }`, and then it could be used:
```
  .then(FileReader.text)
```
or

```
  .then(FileReader.base64)
```
This could also have a default value, so that current calls of `.then(FileReader)` would not need changing...

It won't be quite as neat when it has to take an option.  Any preference?

Reply:
`.then(FileReader.base64)` works for me. Maybe it should be `FileReader.text` and `FileReader.url` though?

I'm not sure about the default option because that means you have to return a function with the `base64` property on it which is always confusing to read.